### PR TITLE
Improve user experience for the storage service

### PR DESCRIPTION
**Pull request by yaxia**
1. Check the source size and make sure it doesn't exceed dest object limitation before data transfer
2. Support AzureChinaCloud environment when the active account is in it

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@yaxia__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (yaxia). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, or work for Microsoft Open Technologies, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

File: lib/commands/storage/storage.blob._js [hunk 1]
```diff
@@ -855,6 +855,17 @@ exports.init = function(cli) {
     if (!fsStatus.isFile()) {
       throw new Error(util.format($('%s is not a file'), specifiedFileName));
     }
+
+    var sizeLimit = StorageUtil.MaxBlockBlobSize;
+    if (specifiedBlobType == BlobConstants.BlobTypes.APPEND) {
```
Comment (yugangw-msft) [inline lib/commands/storage/storage.blob._js]: use ===

File: lib/util/profile/environment.js [hunk 2]
```diff
@@ -237,6 +239,7 @@ Environment.publicEnvironments = [
     commonTenantName: 'common',
     activeDirectoryGraphResourceId: 'https://graph.windows.net/',
     activeDirectoryGraphApiVersion: '2013-04-05',
+    storageEndpointSuffix: '.core.chinacloudapi.cn',
     keyVaultDnsSuffix: '.vault.chinacloudapi.cn'
```
Comment (yugangw-msft) [inline lib/util/profile/environment.js]: Could you let me know what the endpoint for internal dogfood env is? I will update our script

**Comment by yugangw-msft** [comment]
LGTM

**Pull request merged**